test/apps/iodemo/io_demo.cc

```diff
@@ -940,6 +940,11 @@ class DemoServer : public P2pDemoCommon {
         P2pDemoCommon(test_opts), _callback_pool(0, "callbacks") {
     }
 
+    ~DemoServer()
+    {
+        destroy_connections();
```

Comment:
call destroy_listener in run() before exiting
it should cleanup conn requests queue (reject them) and destroy the listener

Reply:
done